### PR TITLE
AFF Generator Shall Generate `$multipleOf` Only for Fields of Type Integer

### PR DESCRIPTION
**Pull request by burcuka**

### CODE REVIEW

**Comment by larshp** [comment]
2500 lines change, suggest moving the indentation changes to a different PR

File: src/zif_aff_log.intf.abap [hunk 1]
```diff
@@ -18,7 +18,8 @@ INTERFACE zif_aff_log
     co_msg124 TYPE string VALUE `Given structure must have mandatory fields format_version and header` ##NO_TEXT,
     co_msg126 TYPE string VALUE `For required fields, a default handling is not possible` ##NO_TEXT,
     co_msg127 TYPE string VALUE `Elements of type enum should be required or have a default` ##NO_TEXT,
-    co_msg128 TYPE string VALUE `Components on top level (except format_version) should be structures or tables` ##NO_TEXT.
+    co_msg128 TYPE string VALUE `Components on top level (except format_version) should be structures or tables` ##NO_TEXT,
+    co_msg129 TYPE string VALUE `Fields of a different type than integer should not use the annototation $multipleOf` ##NO_TEXT.
```
Comment (wurzka) [inline src/zif_aff_log.intf.abap:22]: A warning: @huber-nicolas created messages with number 129 and 130 in his PR #231.
This should be kept in mind when solving merge conflicts

File: src/zcl_aff_writer_json_schema.clas.abap [hunk 2]
```diff
@@ -387,11 +393,8 @@ CLASS zcl_aff_writer_json_schema IMPLEMENTATION.
         min                 = DATA(min_value) ).
     DATA(multiple_of) = abap_doc-multiple_of.
 
-    IF multiple_of IS INITIAL AND element_description->type_kind = cl_abap_typedescr=>typekind_packed.
-      DATA(decimals) = element_description->decimals.
-      IF decimals > 0.
-        multiple_of = |0.{ repeat( val = `0`  occ = decimals - 1 ) }1|.
-      ENDIF.
+    IF multiple_of IS NOT INITIAL AND is_element_descr_kind_int( element_description ) = abap_false.
+      log->add_warning( message_text = zif_aff_log=>co_msg129 component_name = fullname_of_type ).
```
[the remark this answers is not in the document]
Comment (burcuka) [reply src/zcl_aff_writer_json_schema.clas.abap:397]: but in the if statements below we would also write a warning if multiple_of is empty. I think we want to log a warning if its not empty and not an integer. Am I wrong?

[the remark this answers is not in the document]
Comment (wurzka) [reply src/zcl_aff_writer_json_schema.clas.abap:397]: Yes, you're right.

**Comment by schneidermic0** [comment]
I have updated the rules

**Pull request merged**